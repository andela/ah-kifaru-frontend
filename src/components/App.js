import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Routes from '../routes/AppRouter';
import Footer from './Footer/Footer';

const App = () => {
  return (
    <Router>
      <Routes />
      <Footer />
    </Router>
  );
};

export default App;
