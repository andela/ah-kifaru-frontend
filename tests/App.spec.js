import React from 'react';
import Jest from 'jest';
import toJson from 'enzyme-to-json';
import App from '../src/components/App';
import { shallow } from './enzyme';
import LoginPage from '../src/pages/loginPage';
import HomePage from '../src/pages/hompage';
import Routes from '../src/routes/AppRouter';

describe('Application test', () => {
  it('should work fine', () => {
    const wrapper = shallow(<App />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should work fine', () => {
    const wrapper = shallow(<LoginPage />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should work fine', () => {
    const wrapper = shallow(<HomePage />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should work fine', () => {
    const wrapper = shallow(<Routes />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
