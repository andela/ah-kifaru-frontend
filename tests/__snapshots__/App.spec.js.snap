// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application test should work fine Home Page 1`] = `
<div
  className="m-20 text-center"
>
  <h1
    className="text-green-700"
  >
    Welcome to ErrorSwag, what on your mind?
  </h1>
  <img
    alt="errorswag logo"
    src={Object {}}
  />
  <Link
    to="/login"
  >
    <button
      className="bg-green-300 text-white p-2 hover:bg-green-700 border hover:border-red-300"
      type="button"
    >
      Login
    </button>
  </Link>
  <Footer />
</div>
`;

exports[`Application test should work fine on App 1`] = `
<BrowserRouter>
  <Routes />
  <Footer />
</BrowserRouter>
`;

exports[`Application test should work fine on Login Page 1`] = `
<Fragment>
  <Link
    to="/"
  >
    <button
      style={
        Object {
          "backgroundColor": "sky-blue",
          "border": "none",
          "color": "black",
          "marginLeft": "80%",
          "padding": "7px 9px",
        }
      }
      type="button"
    >
      Home
    </button>
  </Link>
  <form>
    <label
      style={
        Object {
          "height": "2em",
          "width": "50%",
        }
      }
    >
      Email
    </label>
    <input
      placeholder="example@email.com"
      style={
        Object {
          "border": "1px solid #333",
          "height": "2em",
          "width": "100%",
        }
      }
      type="email"
    />
    <label
      style={
        Object {
          "height": "2em",
          "width": "50%",
        }
      }
    >
      Password
    </label>
    <input
      placeholder="......."
      style={
        Object {
          "border": "1px solid #333",
          "height": "2em",
          "width": "100%",
        }
      }
      type="password"
    />
    <input
      style={
        Object {
          "backgroundColor": "sky-blue",
          "border": "none",
          "color": "black",
          "marginLeft": "80%",
          "padding": "7px 9px",
        }
      }
      type="submit"
      value="login"
    />
  </form>
</Fragment>
`;

exports[`Application test should work fine on Routes 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/login"
  />
</Switch>
`;
